src/dotnet/HQ.CLI/Commands/TimeEntries/GetTimeEntriesCommand.cs

```diff
@@ -0,0 +1,158 @@
+            }
+            else
+            {
+                startDate = settings.From;
+                endDate = settings.To;
+            }
+
+            var timeEntryRequest = new GetTimesV1.Request
+            {
+                Search = settings.Search,
+                ChargeCode = settings.chargecode,
+                Id = settings.Id,
+                ClientId = settings.ClientId,
+                StaffId = settings.StaffId,
+                StartDate = startDate,
+                EndDate = endDate,
+                SortBy = settings.SortBy,
+                SortDirection = settings.SortDirection,
+                Task = settings.Task,
+                Activity = settings.Activity
+            };
+
+            var result = await _hqService.GetTimeEntriesV1(timeEntryRequest);
+            
+
+            if (!result.IsSuccess || result.Value == null)
+            {
+                ErrorHelper.Display(result);
+                return 1;
+            }
+            
+        OutputHelper.Create(result.Value, result.Value.Records)
+        .WithColumn("ID", t => t.Id.ToString())
+        .WithColumn("DATE", t => t.Date.ToString())
+        .WithColumn("CHARGE CODE", t => t.ChargeCode)
+        .WithColumn("Billable Hours", t => t.BillableHours.ToString())
+        .WithColumn("DESCRIPTION", t => t.Description.Length > 70 ? t.Description.Substring(0, 70) + "..." : t.Description)
+        .WithColumn("ACTIVITY / TASK", t => t.ActivityName != null ? t.ActivityName : t.Task)
+        .WithColumn("REJECTION NOTES", t => t.RejectionNotes)
+        .Output(settings.Output);
+            return 0;
+        }
+
+        private (DateOnly StartDate, DateOnly EndDate) CalculateDateRange(TimePeriod period, DateOnly today)
+        {
+            DateOnly startDate;
+            DateOnly endDate;
+
+            switch (period)
+            {
+                case TimePeriod.Today:
+                    startDate = endDate = today;
+                    break;
+                case TimePeriod.ThisWeek:
+                    startDate = today.AddDays(-(int)today.DayOfWeek);
+                    endDate = startDate.AddDays(6);
+                    break;
+                case TimePeriod.LastWeek:
+                    endDate = today.AddDays(-(int)today.DayOfWeek - 1);
+                    startDate = endDate.AddDays(-6);
+                    break;
+                case TimePeriod.ThisMonth:
+                    startDate = new DateOnly(today.Year, today.Month, 1);
+                    endDate = startDate.AddMonths(1).AddDays(-1);
+                    break;
+                case TimePeriod.LastMonth:
+                    startDate = new DateOnly(today.Year, today.Month, 1).AddMonths(-1);
+                    endDate = startDate.AddMonths(1).AddDays(-1);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            return (startDate, endDate);
+        }
```

Comment:
Let's move these into the `DateOnlyExtensions` class Pri started and add a unit test class for it.